code/ATMOSPHERICS/components/binary_devices/volume_pump.dm

```diff
@@ -28,6 +28,40 @@ Thus, the two variables affect pump operation are set in New():
 	var/id = null
 	var/datum/radio_frequency/radio_connection
 
+/obj/machinery/atmospherics/binary/volume_pump/CtrlClick(mob/living/user)

```

Comment:
Hmm, I'm not sure if you've tested this but if you do this while the area is unpowered, will the pump still turn on?

Reply:
Yep, this could cause issues. I have added sanity checks to prevent this. Thank you.